### PR TITLE
Makisheva/postreview

### PR DESCRIPTION
**Pull request by AlbinaMakisheva**

### CODE REVIEW

File: ui/components/templates/components/Navbar.jinja [hunk 1]
```diff
@@ -17,16 +17,19 @@
...
         <img
           class="ui image left-arrow"
           src="{{ url_for('static', filename='images/arrow-left.png') }}"
           alt="arrow back"
-        />RESTAUROVÁNÍ VŠCHT
+        />
       </button>
-    </a>
 
+      <a class="item" href="/">
```
Comment (mirekys) [inline ui/components/templates/components/Navbar.jinja:29]: Never nest button inside anchor tag - not a valid HTML

File: ui/objects/semantic-ui/js/objects/editform/components/MoreInfo.jsx [hunk 2]
```diff
@@ -57,21 +57,6 @@ export const MoreInfo = ({ activeIndex, handleActive, category }) => {
             }
           />
         </Grid.Column>{" "}
-        <Grid.Column>
-          <VocabularyTreeSelectField
```
Comment (mirekys) [inline ui/objects/semantic-ui/js/objects/editform/components/MoreInfo.jsx:61]: Where did it go? Was it redundant with create form somehow?

**Review by mirekys — CHANGES_REQUESTED** [review]
https://github.com/VSCHT/restaurovani/pull/70/files#diff-bf0825fd3dc894842a62fc9dd9f291015ff7bdec63eaebd4d55f9d57690b3c76R70-R73

Is there any reason to sanitize against different set of valid tags, than what is acceptable by backend?

File: ui/objects/semantic-ui/js/objects/editform/components/MoreInfo.jsx (hunk 2 above)
Comment (AlbinaMakisheva) [reply ui/objects/semantic-ui/js/objects/editform/components/MoreInfo.jsx:61]: According to the review: "	„Další údaje“ – Přijde mi tam zbytečný „typ materiálu“. Ten hlavní vybírám při zakládání karty". Primary material type is selected in CreateForm(kov, sklo, keramika, textil). Secondary material types could be still selected for other materials for the rest of the parts of details of the object

File: ui/branding/semantic-ui/less/default/elements/button.overrides [hunk 3]
```diff
@@ -29,13 +31,20 @@
       margin: 0 auto;
       width: auto;
     }
+    & a {
```
Comment (mirekys) [inline ui/branding/semantic-ui/less/default/elements/button.overrides:34]: anchor tag cannot be nested inside button - produces invalid html

File: ui/components/templates/components/Navbar.jinja [hunk 4]
```diff
@@ -17,16 +17,20 @@
...
           class="ui image left-arrow"
           src="{{ url_for('static', filename='images/arrow-left.png') }}"
           alt="arrow back"
-        />RESTAUROVÁNÍ VŠCHT
+        />
       </button>
-    </a>
 
+      <button class="ui button">
+        <a class="item" href="/">RESTAUROVÁNÍ VŠCHT</a>
```
Comment (mirekys) [inline ui/components/templates/components/Navbar.jinja:30]: Never nest anchor tag inside button - not a valid HTML

**Review by mirekys — CHANGES_REQUESTED** [review]

File: ui/objects/semantic-ui/js/objects/editform/components/RestorationWork.jsx [hunk 5]
```diff
@@ -70,14 +69,14 @@ export const RestorationWork = ({
                 editorConfig={{
                   toolbar:
                     "bold italic | bullist numlist | outdent indent | undo redo",
-                  valid_elements: "strong,b,div,br,p,i,li",
+                  valid_elements:
+                    "a,abbr,acronym,b,blockquote,br,code,div,table,tbody,td,th,tr,em,h1,h2,h3,h4,h5,i,li,ol,p,pre,span,strike,strong,sub,sup,u,ul",
```
Comment (mirekys) [inline ui/objects/semantic-ui/js/objects/editform/components/RestorationWork.jsx:73]: How is it different from what has been merged in oarepo-ui sanitization?

**Review by mirekys — CHANGES_REQUESTED** [review]

File: ui/objects/semantic-ui/js/objects/editform/components/RestorationWork.jsx [hunk 6]
```diff
@@ -70,7 +70,7 @@ export const RestorationWork = ({
                   toolbar:
                     "bold italic | bullist numlist | outdent indent | undo redo",
                   valid_elements:
-                    "a,abbr,acronym,b,blockquote,br,code,div,table,tbody,td,th,tr,em,h1,h2,h3,h4,h5,i,li,ol,p,pre,span,strike,strong,sub,sup,u,ul",
+                    "a[href|title|name|target|rel],abbr[title],acronym[title],b,blockquote,br,code,div,table,tbody,td,th,tr,em,h1,h2,h3,h4,h5,i,li,ol,p,pre,span,strike,strong,sub,sup,u,ul",
```
Comment (mirekys) [inline ui/objects/semantic-ui/js/objects/editform/components/RestorationWork.jsx:73]: Same thing as https://github.com/Narodni-repozitar/nr-docs/pull/117#discussion_r1634395472 and https://github.com/VSCHT/restaurovani/pull/65#discussion_r1636220137

**Review by mirekys — CHANGES_REQUESTED** [review]

**Review by mirekys — CHANGES_REQUESTED** [review]
- [ ] Object detail page not working when navigating from search listing (HTTP 404)
- [ ] Object creation not working (HTTP 404 after filling initial form & clicking create)
- [ ] Invalid date of object record creation:
Screenshot 2024-06-28 at 8 59 10

**Comment by mirekys** [comment]
Plenty of js errors when filling out edit form:
Screenshot 2024-06-28 at 9 12 17

Screenshot 2024-06-28 at 9 12 58

Screenshot 2024-06-28 at 9 13 16

**Review by mirekys — CHANGES_REQUESTED** [review]
https://linear.app/ducesnet/issue/FE-177/fix-ui-layout-of-vedouci-prace-field-on-detail-page still not addressed properly, supervisors hard to read & distinguish name from institution

**Comment by mirekys** [comment]
https://linear.app/ducesnet/issue/EP-111/implement-autosuggest-for-supervisor-and-institution not addressed

**Comment by mirekys** [comment]
Popis input can be resized  out of page
Screenshot 2024-06-28 at 9 57 17

**Comment by AlbinaMakisheva** [comment]
> * [ ]  Object detail page not working when navigating from search listing (HTTP 404)
> * [ ]  Object creation not working (HTTP 404 after filling initial form & clicking create)
> * [ ]  On edit form, clicking `Ulozit` doesn't redirect anywhere (to detail page)
> * [ ]  File upload not working on metadata put (HTTP 404 on: `PUT https://127.0.0.1:5000/api/objects/djmmk-c8408/draft/files/<filename>`
> * [ ]  Invalid date of object record creation:
> 
> Screenshot 2024-06-28 at 8 59 10

redirection to other pages is broken because https://github.com/VSCHT/restaurovani/pull/65#discussion_r1626108767 https://github.com/VSCHT/restaurovani/pull/65#discussion_r1626110729

**Comment by mirekys** [comment]
> redirection to other pages is broken because [#65 (comment)](https://github.com/VSCHT/restaurovani/pull/65#discussion_r1626108767) [#65 (comment)](https://github.com/VSCHT/restaurovani/pull/65#discussion_r1626110729)

That's not the real cause of the problem, that means that value of `edit_html` got somehow broken, @mesemus could you please have a look?

**Comment by mirekys** [comment]
There shouldn't be any "Back" button visible, when on front page

Screenshot 2024-07-09 at 13 26 05

**Comment by corovcam** [comment]
> * [ ]  Object detail page not working when navigating from search listing (HTTP 404)
> * [ ]  Object creation not working (HTTP 404 after filling initial form & clicking create)
> * [ ]  On edit form, clicking `Ulozit` doesn't redirect anywhere (to detail page)
> * [ ]  File upload not working on metadata put (HTTP 404 on: `PUT https://127.0.0.1:5000/api/objects/djmmk-c8408/draft/files/<filename>`
> * [ ]  Invalid date of object record creation:
> 
> Screenshot 2024-06-28 at 8 59 10

1. The file upload still doesn't work. Failes to create a bucket with the given file key, even though the first API call suceeds with 201 CREATED:
  - Request `PUT /api/objects/9j75e-rf686/draft/files/cervenec_2023.pdf` failes with:
  ```
  {"status": 404, "message": "Record '9j75e-rf686' has no file 'cervenec_2023.pdf'."}
  ```

**Review by mirekys — APPROVED** [review]

**Pull request merged**